worker/jobTypes/productionDeployJob.js

```diff
@@ -0,0 +1,182 @@
+  const buildOutput = await workerUtils.promiseTimeoutS(
+    buildTimeout,
+    job.buildRepo(logger),
+    'Timed out on build'
+  );
+  // checkout output of build
+  if (buildOutput && buildOutput.status === 'success') {
+    // only post entire build output to slack if there are warnings
+    const buildOutputToSlack = buildOutput.stdout + '\n\n' + buildOutput.stderr;
+    if (buildOutputToSlack.indexOf('WARNING:') !== -1) {
+      await logger.sendSlackMsg(buildOutputToSlack);
+    }
+
+    return new Promise(function(resolve, reject) {
+      resolve(true);
+    });
+  }
+}
+
+async function pushToProduction(publisher, logger) {
+  const prodOutput = await workerUtils.promiseTimeoutS(
+    buildTimeout,
+    publisher.pushToProduction(logger),
+    'Timed out on push to production'
+  );
+  // checkout output of build
+  if (prodOutput && prodOutput.status === 'success') {
+    await logger.sendSlackMsg(prodOutput.stdout);
+
+    return new Promise(function(resolve, reject) {
+      resolve(true);
+    });
+  }
+}
+
+async function runGithubProdPush(currentJob) {
+  console.log("inside production deploy job")
+  const ispublishable = verifyBranchConfiguredForPublish(currentJob);
+  const userIsEntitled = verifyUserEntitlements(currentJob);
+
+  if (ispublishable === false){
+    workerUtils.logInMongo(currentJob, `${'(BUILD)'.padEnd(15)} You are trying to 
+    
+    to production a branch that is not configured for publishing`)
+    throw new Error('entitlement failed');
+    process.exit
+  }
+  if(userIsEntitled === false){
+    workerUtils.logInMongo(currentJob, `${'(BUILD)'.padEnd(15)} failed, you are not entitled to build or deploy (${repoOwner}/${repoName}) for master branch`);
+    throw new Error('entitlement failed');
+    process.exit
+  }
+
+  workerUtils.logInMongo(currentJob, ' ** Running github push function');
+
+  if (
+    !currentJob ||
+    !currentJob.payload ||
+    !currentJob.payload.repoName ||
+    !currentJob.payload.branchName
+  ) {
+    workerUtils.logInMongo(currentJob,`${'(BUILD)'.padEnd(15)}failed due to insufficient definition`);
+    throw invalidJobDef;
+  }
+
+
+  // TODO: create logging class somewhere else.. for now it's here
+  const Logger = function(currentJob) {
+    return {
+      save: function(message) {
+        workerUtils.logInMongo(currentJob, message);
+      },
+      sendSlackMsg: function(message) {
+        workerUtils.populateCommunicationMessageInMongo(currentJob, message);
+      },
+    };
+  };
+
+  // instantiate github job class and logging class
+  const job = new GitHubJob(currentJob);
+  const logger = new Logger(currentJob);
+  const publisher = new S3Publish(job);
+
+  await startGithubBuild(job, logger);
+
+  console.log('completed build');
+
+  let branchext = '';
+  let isMaster = true;
+
+  if (currentJob.payload.branchName !== 'master') {
+    branchext = '-' + currentJob.payload.branchName;
+    isMaster = false;
+  }
+
+  if (isMaster) {
```

Comment:
we can get rid of any logic around checking for master -- everything moving through this job type is production-bound and as long as the user is entitled to publish it, can go to the same publish calls in S3.